subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultTaskDependency.java

```diff
@@ -130,11 +132,20 @@ private static void addAllFirst(Deque<Object> queue, Object[] items) {
     }
 
     public Set<Object> getValues() {
```

Comment:
Damn, I just realized that we expose this Set directly to the user on the `Task` interface. So users might be depending on the fact that it used to be live. So we might not be able to do this optimization :/